### ISSUE TITLE
libpng version 1.6.43 dll only worked on the debugged version.crashed with the released version.

### ISSUE THREAD

**Issue by lofonder** [pnggroup/libpng#570]
libpng version 1.6.43
zlib version  1.2.11
tools:visual studio 2017
os:win11 x64
target:x86

when I using the 1.6.43 version to compiling the png.dll by using the vs2017.
the debug version of the libpng.dll runned well.but the released version exited with errors.
I dont know why .how to get a vs dll from the png?

heres the tested code with the debuged verion of the libpng.
`
    int      byte;
    int      i, j, k, index, ncolors, bitval, rval, gval, bval, valid;
    int      wpl, d, spp, cindex, tRNS;
    int     png_transforms;
    int    *data, *line, *ppixel;

    int          num_palette, num_text, num_trans;
    png_byte     bit_depth, color_type, channels;
    png_uint_32  w, h, rowbytes, xres, yres;
    png_bytep    rowptr, trans;
    png_bytep   *row_pointers;
    png_structp  png_ptr;
    png_infop    info_ptr, end_info;
    png_colorp   palette;
    png_textp    text_ptr;  /* ptr to text_chunk */

    FILE  *fp = fopen("555.png","rb");
 
    /* Allocate the 3 data structures */
    png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING,(png_voidp)NULL, NULL, NULL);

    info_ptr = png_create_info_struct(png_ptr);

    end_info = png_create_info_struct(png_ptr);

    std::cout<<"fp:"<<fp<<std::endl;
 
    png_init_io(png_ptr, fp);
    std::cout<<"png_ptr:"<<png_ptr<<std::endl;
   //1
    png_transforms = PNG_TRANSFORM_STRIP_16;//PNG_TRANSFORM_IDENTITY ;//PNG_TRANSFORM_STRIP_16;

    std::cout<<"png_transforms:"<<png_transforms<<std::endl;
    /* Read it */
    png_read_png(png_ptr, info_ptr, png_transforms, NULL);
    std::cout<<"png_ptr2222:"<<png_ptr<<std::endl;
    //std::cout<<"info_ptr:"<<info_ptr<<std::endl;
    row_pointers = png_get_rows(png_ptr, info_ptr);

    //std::cout<<"row_pointers:"<<row_pointers<<std::endl;
    w = png_get_image_width(png_ptr, info_ptr);
    h = png_get_image_height(png_ptr, info_ptr);
    bit_depth = png_get_bit_depth(png_ptr, info_ptr);
    rowbytes = png_get_rowbytes(png_ptr, info_ptr);
    color_type = png_get_color_type(png_ptr, info_ptr);
    channels = png_get_channels(png_ptr, info_ptr);
    spp = channels;
    std::cout<<"w:"<<w<<std::endl;
    std::cout<<"h:"<<h<<std::endl;
    std::cout<<"bit_depth:"<<bit_depth<<std::endl;
    std::cout<<"rowbytes:"<<rowbytes<<std::endl;
    std::cout<<"color_type:"<<color_type<<std::endl;
    std::cout<<"channels:"<<channels<<std::endl;
`

**Comment by jbowler** [comment]
> the debug version of the libpng.dll runned well.but the released version exited with errors.

Yeah, but you have to say what the errors were.  Normally this is a symptom of erroneously mixing debug and release DLLs.  VS only supports everything release or everything debug.

**Comment by jbowler** [comment]
@ctruta: just the classic Windows bug of mixing release and debug msvcrt[d].dll.  No comment from @lofonder so close.

**Comment by ctruta** [comment]
> @ctruta: just the classic Windows bug of mixing release and debug msvcrt[d].dll. No comment from @lofonder so close.

Agreed.

**Comment by gememe** [comment]
Just ran into the same issue. I just started using VS2022 and am fairly new to C++ in general. I simply added libpng to my project via vcpkg and everything's working great in debug mode.

But when switching to release mode, calling png_read_info immediately results in `Unhandled exception at ... (ucrtbase.dll) in ....exe: An invalid parameter was passed to a function that considers invalid parameters fatal.`

Any explanation regarding the cause of the issue with possible steps to fix it would be greatly appreciated! Thanks.
